### ISSUE TITLE
New Post Email Notify

### ISSUE THREAD

**Issue by colabsadmin** [r-a-y/buddypress-followers#68]
I added some functionality to my site that will email the authors followers when they publish a new post.  I was using the Subscribe Author plugin, but decided to tap into BP Followers instead.  One less plugin and one less button on the frontend.  The code is here:  https://gist.github.com/colabsadmin/bc54a0e906c6bd8b658b

I'll probably add a BP Notification as well.

**Comment by landwire** [comment]
That's great. So will this live as its own plugin or will it be committed to the buddypress-followers plugin? I guess at the moment I would have to copy the above code into my functions.php or something. Is that right? I assume it will work with any post type? BP Notification would be great...

**Comment by colabsadmin** [comment]
I'm not making a plugin out of it.  I have it in my child theme's functions.php file.  You select what post types to alert on by going into the settings page I've created for it.

**Comment by colabsadmin** [comment]
I was only posting here to make people aware of it.  If you have questions, please post them on gist page since this is NOT supported by r-a-y

**Comment by r-a-y** [comment]
Cool, thanks for posting this, @colabsadmin.

I've added a link to this issue on the wiki.

I think your code is a good start, but I want to be very careful before introducing email notifications.  People will want to turn them off and will want other things added like activity updates, group creation, forum posts, etc.

Screen notifications is tricky at the moment because BuddyPress doesn't have a dedicated screen notifications settings page and some will want to be able to disable screen notifications as well.

**Comment by colabsadmin** [comment]
Totally agree with you.

**Comment by espellcaste** [comment]
That's pretty cool @colabsadmin ! Congrats!

@r-a-y Do you see in the near future a scream notifications settings page as you mentioned in BuddyPress?

**Comment by r-a-y** [comment]
> Do you see in the near future a scream notifications settings page as you mentioned in BuddyPress?

I've created a BP ticket for this: https://buddypress.trac.wordpress.org/ticket/6712.

When I get to screen notifications support for BP Follow, I'll also try and offer backward-compatibility support to BP 1.9.  Depends on how much work I'll have to do for this though ;)

**Comment by espellcaste** [comment]
Understand! =)

**Comment by colabsadmin** [comment]
I've added BP notifications.  I thought about utilizing the follow component that's already registered with  bp followers, but decided to keep it separate.  Its a first pass but seems to be working okay.  I havent added any routines to clear the notifications table if a user is deleted, so in most respects, this is basic code.

**Comment by aswain1** [comment]
Does the BP Notifications component support status updates, or does the user have to actually make a blog post?

**Comment by colabsadmin** [comment]
Its for Posts

**Comment by yumnihon** [comment]
Hi all,
I am new member, i am using plugin buddypress-followers. I get the notify email from followers when they publish a new post in admin wordpress.
How to get the notify email from followers when they publish a new Activity on their Activity Stream.

**Comment by Nrddy** [comment]
Hoping you guys are still on this... do you know if it works for custom post types too? I added the code and everything but it doesn't seem to be working for my custom post type called casting-call. no worries if its not.

**Comment by IudazaRoadrunner** [comment]
Hy colabsadmin. I added this code in functions.php but it doesn't work I have a custom post caled "video" http://i.imgur.com/pScQpDX.png

**Comment by Martin128** [comment]
It works for me but just from back end. It is possible to fix?

**Comment by zzbeck** [comment]
Anyone know if this is still possible? I can't find a working solution anywhere.

**Comment by mrjarbenne** [comment]
Have you tried the snippet? Others seem to be using it successfully. An addition to the functions.php file can be easily accomplished via a child theme. 

https://developer.wordpress.org/themes/advanced-topics/child-themes/#how-to-create-a-child-theme

Otherwise, there are a number of plugins that accomplish an email notification upon new post creation. Jetpack has a subscribe by email functionality. Postmatic does as well if you want to avoid  jetpack.

**Comment by zzbeck** [comment]
Yes, I have tried the snippet. I don't know if it's because I'm using WP Multisite, that It won't work.

**Comment by mrjarbenne** [comment]
If you want it to work on multiple sites, you might want to convert the snippet to a plugin. Code for a functions.php file is basically a plugin without the requisite header content. Here’s a tutorial on creating a functionality plugin. 

http://wpcandy.com/teaches/how-to-create-a-functionality-plugin/#.WnJLXHByaEc
